core/BoidsGenerator.cpp

```diff
@@ -10,22 +10,42 @@ void Core::BoidsGenerator::generateBoids()
 {
     int boxHalfSize = m_boxSize / 2;
 
-    for(int i = 0; i < m_numEntities; ++i) 
+    for (auto &phy_boids : m_phy_boids)
     {
-        float rx = static_cast<float> (rand()) / static_cast<float> (RAND_MAX);
-        float ry = static_cast<float> (rand()) / static_cast<float> (RAND_MAX);
-        float rz = static_cast<float> (rand()) / static_cast<float> (RAND_MAX);
+        float x = boxHalfSize * (2 * randomZeroOne() - 1.f);
+        float y = boxHalfSize * (2 * randomZeroOne() - 1.f);
+        float z = boxHalfSize * (2 * randomZeroOne() - 1.f);
+        //float vlim = (randomZeroOne()-0.5f)*10;
+        float vlim = 1.0f;
+        float vx = randomZeroOne() - 0.5f;
```

Comment:
Have a look at DiligentGraphics/Math.hpp something.. in third_parties. You can use Math::float3 and else to improve the code, instead to play float by float. Function length( ) also present in this lib will give you the norm3D that you implemented here, hence this one can be removed.